AccessTokenProvider (#135)
propeldata/ui-kit: app/storybook/documentation/ui-kit/provider/AccessTokenProvider.mdx

import { Canvas, ArgTypes, Meta, Story, Source, Description, Primary, Controls, Stories } from '@storybook/blocks'
import { prettier } from '../../../src/utils'
import * as AccessTokenProviderStories from '../../../../../packages/ui-kit/src/components/AccessTokenProvider/AccessToken.stories'

<Meta title="PROVIDERS/AccessTokenProvider" /> 

# AccessTokenProvider

> The `AccessTokenProvider` allows you to provide a function that fetches an access token from your backend. By using this function, the provider will serve the fetched access token to all of its children components and will automatically refresh the token when it expires.

Reviewer: ```suggestion
The `AccessTokenProvider` allows you to provide a function that fetches an access token from your backend. By using this function, the provider will serve the fetched access token to all of its child components and will automatically refresh the token when it expires.
```

This way, instead of manually providing the access token to each component like this:

```jsx
import React, { useState, useEffect } from 'react'

import { Counter, TimeSeries, Leaderboard } from "@propeldata/ui-kit"  

import { fetchToken } from './utils'

function App() {
  const [accessToken, setAccessToken] = useState(null)

  useEffect(() => {
    async function fetchAccessToken() {
      const accessToken = await fetchToken()
      setAccessToken(accessToken)
    }

    fetchAccessToken()
  }, [fetchToken, setAccessToken])

  return (  
    <Counter query={{ accessToken }} />
    <TimeSeries query={{ accessToken }} />
    <Leaderboard query={{ accessToken }} />
  );  
}
```

> You can just wrap your app with the `AccessTokenProvider`, pass it the `fetchToken` function and it will automatically provide the access token to all of its child components:

Reviewer: ```suggestion
You can just wrap your app with the `AccessTokenProvider`, pass it the `fetchToken` function, and it will automatically provide the access token to all of its child components:
```

> You can just wrap your app with the `AccessTokenProvider`, pass it the `fetchToken` function and it will automatically provide the access token to all of its children components:

Reviewer: ```suggestion
You can just wrap your app with the `AccessTokenProvider`, pass it the `fetchToken` function and it will automatically provide the access token to all of its child components:
```

```jsx
import React from 'react'

import { Counter, TimeSeries, Leaderboard, AccessTokenProvider } from "@propeldata/ui-kit"  

import { fetchToken } from './utils'

function App() {
  return (
    <AccessTokenProvider fetchToken={fetchToken}>
      <Counter />
      <TimeSeries />
      <Leaderboard />
    </AccessTokenProvider>
  );  
}
```

> If you'd prefer to handle fetching on your own, you can pass the `accessToken` prop to the `AccessTokenProvider` and an `onAccessTokenExpired` callback to handle token expiration. The provided accessToken will be served to all of its children components:

Reviewer: ```suggestion
If you'd prefer to handle fetching on your own, you can pass the `accessToken` prop to the `AccessTokenProvider` and an `onAccessTokenExpired` callback to handle token expiration. The provided access token will be served to all of its child components:
```
